lib/puppet/parser/scope.rb

```diff
@@ -738,10 +815,12 @@ def to_s
 
   alias_method :inspect, :to_s
 
-  # remove ephemeral scope up to level
-  # TODO: Who uses :all ? Remove ??
+  # Pop ephemeral scopes up to level and return them
   #
+  # @deprecated use #pop_epehemeral
+  # @api private
   def unset_ephemeral_var(level=:all)
+    Puppet.deprecation_warning('Method Parser::Scope#unset_ephemeral_var() is deprecated')
```

Comment:
It is only unset_ephemeral_var(:all) that is deprecated. This line needs to go inside the if below. And it should mention :alll in the error. (We do not want people to have to make changes to use pop_ephemerals, and later have to change again when we change how scope works). It is simply the :all case that we know will no longer be supported (and is causing potential harm if used).